diff -r 90d027eb635f src/Makefile
--- a/src/Makefile	Sun Apr 30 22:14:00 2017 -0400
+++ b/src/Makefile	Fri Jun 02 16:30:04 2017 -0700
@@ -34,11 +34,17 @@
   libluajit = luajit/src/lua51.dll
 else ifeq (osx, $(findstring osx, $(MAKECMDGOALS)))
   # Cross-compile for Mac OSX.
-  CROSS = i686-apple-darwin10-
-  CFLAGS = -m32 -arch i386 -mdynamic-no-pic -mmacosx-version-min=10.5 \
-           -isysroot /usr/lib/apple/SDKs/MacOSX10.5.sdk -Os
-  CXXFLAGS = -m32 -arch i386 -mdynamic-no-pic -mmacosx-version-min=10.5 \
+  ifeq (Darwin, $(shell uname -s))
+    CROSS =
+    CFLAGS = -mdynamic-no-pic -D_DARWIN_C_SOURCE -I/usr/local/Cellar/ncurses/6.0_3/include -Os
+    CXXFLAGS = -mdynamic-no-pic -D_DARWIN_C_SOURCE -I/usr/local/Cellar/ncurses/6.0_3/include -Os
+  else
+    CROSS = i686-apple-darwin10-
+    CFLAGS = -m32 -arch i386 -mdynamic-no-pic -mmacosx-version-min=10.5 \
              -isysroot /usr/lib/apple/SDKs/MacOSX10.5.sdk -Os
+    CXXFLAGS = -m32 -arch i386 -mdynamic-no-pic -mmacosx-version-min=10.5 \
+               -isysroot /usr/lib/apple/SDKs/MacOSX10.5.sdk -Os
+  endif
   LUA_CFLAGS = -DLUA_USE_MACOSX
   LDFLAGS = -liconv -rdynamic
   MAKE = make
@@ -56,7 +62,11 @@
                           --cflags glib-2.0)
   else
     plat_flag = -DCURSES -D_XOPEN_SOURCE_EXTENDED
-    CURSES_LIBS = -lncurses
+    ifeq (Darwin, $(shell uname -s))
+      CURSES_LIBS = -L/usr/local/Cellar/ncurses/6.0_3/lib -lncurses
+    else
+      CURSES_LIBS = -lncurses
+    endif
   endif
   libluajit = luajit/src/libluajit.osx.a
 else
@@ -139,9 +149,11 @@
 termkey_unix_objs = driver-ti.o driver-csi.o
 termkey_win32_objs = driver-win32-pdcurses.o
 termkey_objs = termkey.o $(termkey_unix_objs) $(termkey_win32_objs)
-cdk_objs = binding.o buttonbox.o cdk.o cdk_display.o cdk_objs.o cdkscreen.o \
+# Need to prefix these b/c OS X filenames are case-insensitive and selection.o conflicts with scintilla's.
+cdk_objs_base = binding.o buttonbox.o cdk.o cdk_display.o cdk_objs.o cdkscreen.o \
            draw.o entry.o fselect.o itemlist.o label.o mentry.o popup_label.o \
            scroll.o scroller.o select_file.o selection.o traverse.o version.o
+cdk_objs = $(addprefix cdk-, $(cdk_objs_base))

 # Add debugging symbols and disable optimizations when DEBUG=1.
 # Note: In order to profile with gprof (-pg), do not retain symbols in LDFLAGS.
@@ -193,7 +205,7 @@
 $(gtdialog_objs): gtdialog/gtdialog.c
 	$(CROSS)$(CC) -c $(CFLAGS) -std=c99 -pedantic -D_POSIX_C_SOURCE=200809L \
 		$(gtdialog_flags) $< -o $@
-$(cdk_objs): %.o: cdk/%.c
+$(cdk_objs): cdk-%.o: cdk/%.c
 	$(CROSS)$(CC) -c $(CFLAGS) -D_GNU_SOURCE -Itermkey -Icdk $(CURSES_CFLAGS) $< \
 		-o $@
 $(termkey_objs): %.o: termkey/%.c
diff -r 90d027eb635f src/textadept.c
--- a/src/textadept.c	Sun Apr 30 22:14:00 2017 -0400
+++ b/src/textadept.c	Fri Jun 02 16:30:04 2017 -0700
@@ -2488,7 +2488,12 @@
   uint32_t size = FILENAME_MAX;
   _NSGetExecutablePath(path, &size);
   textadept_home = realpath(path, NULL);
-  p = strstr(textadept_home, "MacOS"), strcpy(p, "Resources\0");
+  p = strstr(textadept_home, "MacOS");
+  if (p != NULL) {
+    strcpy(p, "Resources\0");
+  } else {
+    if ((last_slash = strrchr(textadept_home, '/'))) *last_slash = '\0';
+  }
   free(path);
 #if !CURSES
   osxapp = g_object_new(GTKOSX_TYPE_APPLICATION, NULL);
diff -r 90d027eb635f src/termkey.patch
--- a/src/termkey.patch	Sun Apr 30 22:14:00 2017 -0400
+++ b/src/termkey.patch	Fri Jun 02 17:27:28 2017 -0700
@@ -216,7 +216,7 @@
    if(tk->fd != -1 && !(tk->flags & TERMKEY_FLAG_NOTERMIOS)) {
      struct termios termios;
      if(tcgetattr(tk->fd, &termios) == 0) {
-@@ -472,9 +481,9 @@
+@@ -472,9 +481,10 @@
          /* want no signal keys at all, so just disable ISIG */
          termios.c_lflag &= ~ISIG;
        else {
@@ -225,6 +225,7 @@
          termios.c_cc[VQUIT] = _POSIX_VDISABLE;
 -        termios.c_cc[VSUSP] = _POSIX_VDISABLE;
 +        termios.c_cc[VINTR] = _POSIX_VDISABLE;
++        termios.c_cc[VLNEXT] = _POSIX_VDISABLE;
          /* Some OSes have Ctrl-Y==VDSUSP */
  #ifdef VDSUSP
          termios.c_cc[VDSUSP] = _POSIX_VDISABLE;
