diff -r e2c08dc37617 src/Makefile
--- a/src/Makefile	Wed May 02 11:01:52 2018 -0400
+++ b/src/Makefile	Wed May 02 08:51:30 2018 -0700
@@ -36,7 +36,11 @@
   endif
 else ifeq (osx, $(findstring osx, $(MAKECMDGOALS)))
   # Cross-compile for Mac OSX.
-  CROSS = x86_64-apple-darwin9-
+  ifeq (Darwin, $(shell uname -s))
+    CROSS =
+  else
+    CROSS = x86_64-apple-darwin9-
+  endif
   CFLAGS += -mdynamic-no-pic
   CXXFLAGS += -mdynamic-no-pic
   LUA_CFLAGS = -DLUA_USE_MACOSX
@@ -56,6 +60,11 @@
   else
     plat_flag = -DCURSES
     CURSES_LIBS = -lncurses
+    ifeq (Darwin, $(shell uname -s))
+      CURSES_LIBS += -L/usr/local/Cellar/ncurses/6.0_3/lib
+      CFLAGS += -D_DARWIN_C_SOURCE -I/usr/local/Cellar/ncurses/6.0_3/include
+      CXXFLAGS += -D_DARWIN_C_SOURCE -I/usr/local/Cellar/ncurses/6.0_3/include
+    endif
   endif
 else
   # Build for Linux/BSD.
diff -r e2c08dc37617 src/textadept.c
--- a/src/textadept.c	Wed May 02 11:01:52 2018 -0400
+++ b/src/textadept.c	Wed May 02 08:51:30 2018 -0700
@@ -2456,7 +2456,12 @@
   uint32_t size = FILENAME_MAX;
   _NSGetExecutablePath(path, &size);
   textadept_home = realpath(path, NULL);
-  p = strstr(textadept_home, "MacOS"), strcpy(p, "Resources\0");
+  p = strstr(textadept_home, "MacOS");
+  if (p != NULL) {
+    strcpy(p, "Resources\0");
+  } else {
+    if ((last_slash = strrchr(textadept_home, '/'))) *last_slash = '\0';
+  }
   free(path);
 #if !CURSES
   osxapp = g_object_new(GTKOSX_TYPE_APPLICATION, NULL);
diff -r e2c08dc37617 src/termkey.patch
--- a/src/termkey.patch	Wed May 02 11:01:52 2018 -0400
+++ b/src/termkey.patch	Wed May 02 08:51:30 2018 -0700
@@ -218,7 +218,7 @@
    if(tk->fd != -1 && !(tk->flags & TERMKEY_FLAG_NOTERMIOS)) {
      struct termios termios;
      if(tcgetattr(tk->fd, &termios) == 0) {
-@@ -472,9 +481,9 @@
+@@ -472,9 +481,10 @@
          /* want no signal keys at all, so just disable ISIG */
          termios.c_lflag &= ~ISIG;
        else {
@@ -227,6 +227,7 @@
          termios.c_cc[VQUIT] = _POSIX_VDISABLE;
 -        termios.c_cc[VSUSP] = _POSIX_VDISABLE;
 +        termios.c_cc[VINTR] = _POSIX_VDISABLE;
++        termios.c_cc[VLNEXT] = _POSIX_VDISABLE;
          /* Some OSes have Ctrl-Y==VDSUSP */
  #ifdef VDSUSP
          termios.c_cc[VDSUSP] = _POSIX_VDISABLE;
